packages/@aws-cdk/aws-ecr-assets/lib/image-asset.ts

```diff
@@ -93,10 +93,24 @@ export class DockerImageAsset extends Construct implements assets.IAsset {
       exclude = [...exclude, ...fs.readFileSync(ignore).toString().split('\n').filter(e => !!e)];
     }
 
+    // include build context in "extra" so it will impact the hash
+    const extraHash = new Array<any>();
+    if (props.extra) { extraHash.push(props.extra); }
+    if (props.buildArgs) {
+      extraHash.push({ buildArgs: props.buildArgs });
+    }
+    if (props.target) {
+      extraHash.push({ target: props.target });
+    }
+    if (props.file) {
+      extraHash.push({ file: props.file });
+    }
+
     const staging = new assets.Staging(this, 'Staging', {
       ...props,
       exclude,
-      sourcePath: dir
+      sourcePath: dir,
+      extra: extraHash.length === 0 ? undefined : extraHash.map(x => JSON.stringify(x)).join('=====')
```

Comment:
Why not just make ExtraHash a stringified JSON object? What is the added value of it being an array here?

Reply:
Done damn you